blueman/gui/GenericList.py

```diff
@@ -101,54 +94,32 @@ def prepend(self, **columns: object) -> Gtk.TreeIter:
         vals = self._add(**columns)
         return self.liststore.prepend(vals)
 
-    def get_conditional(self, **cols: object) -> List[int]:
+    def get_conditional(self, **cols: object) -> List[Gtk.TreeIter]:
         ret = []
-        matches = 0
-        for i in range(len(self.liststore)):
-            row = self.get(i)
-            for k, v in cols.items():
-                if row[k] == v:
-                    matches += 1
-
-            if matches == len(cols):
-                ret.append(i)
-
-            matches = 0
+        for tree_row in self.liststore:
+            row_data = self.get(tree_row.iter)
+            for key in cols:
+                if row_data[key] == cols[key]:
+                    ret.append(tree_row.iter)
```

Comment:
This seems wrong to me.

We used to append a single entry for the row if `all(row_data[k] == v for (k, v) in cols.items())` (one for each row if all columns match). Now we're appending the row multiple times for each matching column. So basically this changes from logical AND to logical OR and duplicates results.

It's not actually a problem as the only use case seems to be with only the `name` key and the result is the same in case of a single column. Plus, the use case only uses the first result, so there's also no point in collecting at all.

We should probably just strip it down to something like this directly in `PluginDialog`:

```python
for tree_row in self.list.liststore:
    if self.list.get(tree_row.iter)["name"] == name:
        self.list.set(tree_row.iter, active=loaded)
        break
```